lxd/patches.go

```diff
@@ -2247,6 +2248,60 @@ func patchStorageApiInsertZfsDriver(name string, d *Daemon) error {
 	return nil
 }
 
+func patchStorageZFSnoauto(name string, d *Daemon) error {
+	pools, err := dbStoragePools(d.db)
+	if err != nil {
+		if err == NoSuchObjectError {
+			return nil
+		}
+		logger.Errorf("Failed to query database: %s", err)
+		return err
+	}
+
+	for _, poolName := range pools {
+		_, pool, err := dbStoragePoolGet(d.db, poolName)
+		if err != nil {
+			logger.Errorf("Failed to query database: %s", err)
+			return err
+		}
+
+		if pool.Driver != "zfs" {
+			continue
+		}
+
+		zpool := pool.Config["zfs.pool_name"]
+		if zpool == "" {
+			continue
+		}
+		paths := []string{fmt.Sprintf("%s/containers", zpool), fmt.Sprintf("%s/custom", zpool)}
```

Comment:
A temporary workaround is to stop the current upgrade and:

```
systemctl stop lxd.service lxd.socket
zfs create -p -o mountpoint=none rpool/virtual/lxd/custom
```
and try to upgrade again. This could take a little while.